esmvaltool/diag_scripts/weathertyping/calc_utils.py

```diff
@@ -0,0 +1,1095 @@
+    wt_cubes
+        List of cubes of lwt, slwt_ERA5 and slwt_EOBS
+    data_info
+        Dictionary with info to dataset
+    """
+    var_name = data_info.get("var")
+    wt_string = data_info.get("wt_string")
+    driver = get_driver(data_info)
+
+    logger.info(
+        "Calculating %s %s means for %s",
+        data_info.get("dataset"),
+        var_name,
+        wt_string,
+    )
+
+    wt_array, tcoord = get_wt_array(wt_string, wt_cubes)
+
+    for wt in range(1, len(np.unique(wt_array)) + 1):
+        target_indices = np.where(wt_array == wt)
+        if len(target_indices[0]) < 1:
+            logger.info(
+                "calc_wt_means - CAUTION: Skipped %s %s \
+                for dataset %s!",
+                wt_string,
+                wt,
+                data_info.get("dataset"),
+            )
+            continue
+        dates = [
+            tcoord.units.num2date(tcoord.points[i]) for i in target_indices
+        ]
+        extracted_cube = cube.extract(
+            iris.Constraint(time=lambda t, d=dates: t.point in d[0]),
+        )
+        wt_cube_mean = extracted_cube.collapsed("time", iris.analysis.MEAN)
+        plot_maps(wt, cfg, wt_cube_mean, data_info, "mean")
+
+    ancestors = [
+        f"{data_info.get('preproc_path')}",
+        f"{cfg.get('work_dir')}/ERA5.nc",
+    ]
+    provenance_record = get_provenance_record(
+        f"{var_name} means for {wt_string}",
+        ancestors,
+        [var_name],
+        ["map"],
+        ["mean"],
+    )
+
+    local_path = f"{cfg.get('plot_dir')}/mean"
+
+    log_provenance(
+        f"{local_path}/{wt_string}_{wt}{driver}_"
+        f"{data_info.get('dataset')}_{data_info.get('ensemble', '')}"
+        f"_{var_name}_mean_{data_info.get('timerange')}",
+        cfg,
+        provenance_record,
+    )
+
+
+def get_wt_array(wt_string: str, wt_cubes: iris.cube.CubeList) -> tuple:
+    """Get weathertype array and time coordinate based on wt_string.
+
+    Parameters
+    ----------
+    wt_string
+        string for weathertype selection
+    wt_cubes
+        list of weathertype cubes
+
+    Raises
+    ------
+    NameError
+        if wt_array does not exist for the given wt_string
+
+    Returns
+    -------
+    tuple
+        weathertype array and time coordinate
+    """
+    if wt_string == "slwt_ERA5":
+        slwt_era5_cube = wt_cubes[1]
+        tcoord = slwt_era5_cube.coord("time")
+        wt_array = slwt_era5_cube.data[:]
+    elif wt_string == "slwt_EOBS":
+        slwt_eobs_cube = wt_cubes[2]
+        tcoord = slwt_eobs_cube.coord("time")
+        wt_array = slwt_eobs_cube.data[:]
+    elif wt_string == "lwt":
+        lwt_cube = wt_cubes[0]
+        tcoord = lwt_cube.coord("time")
+        wt_array = lwt_cube.data[:]
+    else:
+        error_str = "wt_array does not exist for calc_wt_means, calc_wt_anomalies or calc_wt_std."
+        raise NameError(
+            error_str,
+        )
+
+    return wt_array, tcoord
```

Comment:
it'd be useful if you could return and further process `wt_array` as a lazy object ie use the `,core_data()` member of the iris cube